src/meili/candidate_search.cc

```diff
@@ -70,7 +72,7 @@ CandidateCollector::WithinSquaredDistance(const midgard::PointLL& location,
     float offset;
 
     baldr::GraphId snapped_node;
-    baldr::PathLocation correlated(baldr::Location(location, baldr::Location::StopType::BREAK));
+    baldr::PathLocation correlated(baldr::Location(location, stop_type));
```

Comment:
@kevinkreiser so I think this change makes a lot of sense. was this just a latent bug or is it required for the rest of the change here to work.

Reply:
it is required! a few things are going on here:

1. map matching was already returning uturn routes but it was penalizing them and downstream from the match we papered over any uturn by not trimming the path edge to the uturn spot
2. map matching never cared what type of location you were using before (which is why it was ok to just default to BREAK here).
3. this change and related other ones allow the correct type to be passed through to the map matcher so that it can use it to determine if it should be allowed to do a uturn at all (at the trace point).

basically the `*through*` types of points mean that you must continue in the direction you came from both for routing and now for map matching. if you didnt select a through or break through type then you are telling the router and now the matcher that you are cool with doing a uturn if that makes the most sense for the route/match. have a read of the `type` documentation https://github.com/valhalla/valhalla/blob/master/docs/api/turn-by-turn/api-reference.md#locations. it explains the 4 values and how they encapsulate two binary proprites (allowing uturn, building a leg).